.github/workflows/test.reusable.yml

```diff
@@ -25,49 +29,77 @@ jobs:
             ~/go/pkg/mod
             ~/go/bin
             ~/.cache
-          key: ${{ runner.os }}-amd64-go-${{ hashFiles('**/go.sum') }}
-          restore-keys: ${{ runner.os }}-amd64-go-
+          key: ${{ runner.os }}-${{ inputs.arch }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: ${{ runner.os }}-${{ inputs.arch }}-go-
 
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
           go-version: ${{ inputs.go-version }}
 
-      - name: Setup go-acc
-        run: go install github.com/ory/go-acc@latest
+      - name: Docker Setup QEMU
+        uses: docker/setup-qemu-action@v2.1.0
+        if: inputs.arch != 'amd64'
 
-      - name: Set up gotestfmt
-        uses: haveyoudebuggedit/gotestfmt-action@v2
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }} # Avoid getting rate limited
+      - name: Run test
+        env:
+          CGO_ENABLED: "1"
+          TEST_BENCH_OPTION: -bench=.
+          TEST_FLAGS: ""
+        run: |2
+          docker run \
+            --platform linux/${{ inputs.arch }} \
+            --env CGO_ENABLED \
+            --env TEST_BENCH_OPTION \
+            --workdir ${{ github.workspace }} \
+            --volume /home/runner:/home/runner \
+            --interactive \
+            golang:${{ inputs.go-version }}-buster \
+            bash -s <<'EOF'
+              set -eo pipefail
 
```

Comment:
Would be nice to print the current environment (`uname -a` maybe?)

Reply:
Done :)